test/WebJobs.Script.Tests.Integration/Host/ScriptHostManagerTests.cs

```diff
@@ -37,88 +37,119 @@ public ScriptHostManagerTests()
         [Fact]
         public async Task UpdateFileAndRestart()
         {
-            CancellationTokenSource cts = new CancellationTokenSource();
-            var fixture = new NodeEndToEndTests.TestFixture();
-            var blob1 = await UpdateOutputName("testblob", "first", fixture);
-
-            await fixture.Host.StopAsync();
+            var fixture = new NodeEndToEndTests.TestFixture(false);
             var config = fixture.Host.ScriptConfig;
 
-            ExceptionDispatchInfo exception = null;
+            config.OnConfigurationApplied = c =>
+            {
+                c.Functions = new Collection<string> { "TimerTrigger" };
+            };
+
+            var blob1 = await UpdateOutputName("testblob", "first", fixture);
+
             using (var eventManager = new ScriptEventManager())
             using (var manager = new ScriptHostManager(config, eventManager))
             {
-                // Background task to run while the main thread is pumping events at RunAndBlock().
-                Thread t = new Thread(_ =>
+                string GetErrorTraces()
                 {
-                    // don't start until the manager is running
-                    TestHelpers.Await(() => manager.State == ScriptHostState.Running).Wait();
+                    var messages = fixture.LoggerProvider.GetAllLogMessages()
+                        .Where(t => t.Level == LogLevel.Error)
+                        .Select(t => t.FormattedMessage);
+
+                    return string.Join(Environment.NewLine, messages);
+                }
 
+                List<Exception> exceptions = new List<Exception>();
+
+                // Background task to run while the main thread is pumping events at RunAndBlock().
+                Thread backgroundThread = new Thread(_ =>
+                {
                     try
                     {
+                        // don't start until the manager is running
+                        TestHelpers.Await(() => manager.State == ScriptHostState.Running,
+                            userMessageCallback: GetErrorTraces).GetAwaiter().GetResult();
+
                         // Wait for initial execution.
-                        TestHelpers.Await(async () =>
-                        {
-                            bool exists = await blob1.ExistsAsync();
-                            return exists;
-                        }, timeout: 10 * 1000).Wait();
+                        TestHelpers.Await(() =>
+                         {
+                             bool exists = blob1.ExistsAsync().GetAwaiter().GetResult();
```

Comment:
nit: this and the other call below (in the other lambda) should be made async

Reply:
Missed these somehow. I'll fix.